### PR TITLE
[ML] Preserve alias attributes when making them hidden

### PR DESCRIPTION
**Pull request by przemekwitek**
This is a followup to #77416.

If aliases on ML hidden indices are made hidden then
all attributes of the alias need to be preserved.

The particularly critical case for ML is where a job's
results alias has a filter that restricts documents
returned when searching the shared results index to
those that relate to the job.

### CODE REVIEW

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlInitializationServiceIT.java [hunk 1]
```diff
@@ -114,6 +113,113 @@ public void testThatMlIndicesBecomeHiddenWhenTheNodeBecomesMaster() throws Excep
...
+        List<String> mlHiddenIndexNames = List.of(
+            ".ml-anomalies-7",
+            ".ml-state-000001",
+            ".ml-stats-000001",
+            ".ml-notifications-000002",
+            ".ml-annotations-000001"
+        );
+        for (String indexName : mlHiddenIndexNames) {
+            try {
+                createIndex(indexName, Settings.builder().put(SETTING_INDEX_HIDDEN, false /*randomBoolean()*/).build());
```
Comment (droberts195) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlInitializationServiceIT.java:126]: Did you intend to uncomment `randomBoolean()`?

**Review by droberts195 — APPROVED** [review]
LGTM apart from one debugging leftover.

But before you merge this please can you run a few more iterations of BWC on #79123 with the `continue` condition that I messed up corrected.  Then we can be sure that the code really does work when upgrading a real 6.8 cluster to 7.16.

**Comment by przemekwitek** [comment]
> But before you merge this please can you run a few more iterations of BWC on #79123 with the continue condition that I messed up corrected. Then we can be sure that the code really does work when upgrading a real 6.8 cluster to 7.16.

Sure, will do.

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlInitializationServiceIT.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlInitializationServiceIT.java]: Done.

**Pull request merged**